### PR TITLE
Set bucket, path & createdAt default values on test artifacts

### PR DESCRIPTION
**Pull request by makesitgo**

### CODE REVIEW

**Comment by makesitgo** [comment]
@tychoish  ... not sure how you prefer to manager PRs, commit messages, etc. but I'm all ears to tweak this however need be.

With regards to the actual changes here, I don't _love_ adding the `conf` arg to the `Validate`/`Convert` functions, couldn't find many other ways to gracefully pass down bucket configuration details down to the TestArtifact.  Again, all ears here on suggestions if you see any alternative approaches here.

**Comment by makesitgo** [comment]
I also can't seem to request a reviewer on this (likely some Github access control), but if you could take a look it'd be much appreciated 😬

File: report.go [hunk 1]
```diff
@@ -88,9 +88,13 @@ type TestArtifact struct {
 
 // Validate examines an entire artifact structure and reports if there
 // are any logical inconsistencies with the data.
-func (a *TestArtifact) Validate() error {
+func (a *TestArtifact) Validate(conf BucketConfiguration) error {
```
Comment (tychoish) [inline report.go]: I'd avoid passing conf here, and just add a validate method to the report at the top level to set this default (and maybe validate all the artifacts?) 

There are some methods on the grip.Catcher interface that would help in writing that function.

File: harness_results.go [hunk 2]
```diff
@@ -52,6 +52,8 @@ func (res *BenchmarkResult) Export() Test {
 		Artifacts: []TestArtifact{
 			{
 				LocalFile:        res.ArtifactPath,
+				Path:             res.ArtifactPath,
```
Comment (tychoish) [inline harness_results.go]: I dug in a bit more looking at how files get uploaded, and I have a couple of thoughts/concerns here. 

First, we (might?) want to call filepath.Base() on this string, if users specify a filename with a lot of depth or an absolute path? It's probably safe, but I'm not sure. The inverse is probably also a concern... what if people end up having filenames with not enough information and later test runs override earlier ones?

Second, it seems like the Upload method on TestArtifact might be a good place to populate this value if it's not set. I realize this is counter to my earlier comment, but I don't think there's any good reason for that method to be so aggressive.

**Comment by tychoish** [comment]
We should probably make a MAKE ticket for this change, and just put the ticket name in the PR.

**Comment by makesitgo** [comment]
Closing this in favor of https://github.com/evergreen-ci/poplar/pull/29

**Pull request closed without merging**